pyprocessor/utils/core/cache_manager.py

```
@@ -0,0 +1,1031 @@
+    def _estimate_size(self, value: Any) -> int:
+        """Estimate the size of a value in bytes."""
+        try:
+            # Try to get the size using sys.getsizeof
+            import sys
+
+            return sys.getsizeof(value)
+        except (ImportError, TypeError):
+            # Fallback to a rough estimate based on pickle size
+            try:
+                return len(pickle.dumps(value))
+            except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)